File: 2015/03/one.py
data = open("input.txt").read()
visited = {(0,0): 1}
x = 0
y = 0
for c in data:
    if c == ">":
        x += 1
    elif c == "<":
        x -= 1
    elif c == "v":
        y += 1
    elif c == "^":
        y -= 1

    if (x, y) not in visited:
        visited[(x, y)] = 0
    
    visited[(x, y)] += 1

print(len(visited))

File: 2016/01/one.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	px, py := 0, 0
	mx, my := 0, 1

	data, err := os.ReadFile("input.txt")
	if err != nil { panic(err) }

	input := string(data)
	parts := strings.Split(input, ", ")

	for _, part := range parts {

		if part[0] == 'R' {
			mx, my = my, -mx
		} else {
			mx, my = -my, mx
		}

		num, err := strconv.Atoi(part[1:])
		if err != nil { panic(err) }

		px = px + mx * num
		py = py + my * num
	}
	
	fmt.Println(max(px, -px) + max(py, -py))
}